com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Drawers.cs

```diff
@@ -298,14 +303,33 @@ static void Drawer_PhysicalCamera(SerializedHDCamera p, Editor owner)
             using (new EditorGUI.IndentLevelScope())
             {
                 EditorGUI.BeginChangeCheck();
-                int filmGateIndex = Array.IndexOf(k_ApertureFormatValues, new Vector2((float)Math.Round(cam.sensorSize.vector2Value.x, 3), (float)Math.Round(cam.sensorSize.vector2Value.y, 3)));
-                if (filmGateIndex == -1)
-                    filmGateIndex = EditorGUILayout.Popup(cameraTypeContent, k_ApertureFormatNames.Length - 1, k_ApertureFormatNames);
-                else
-                    filmGateIndex = EditorGUILayout.Popup(cameraTypeContent, filmGateIndex, k_ApertureFormatNames);
 
-                if (EditorGUI.EndChangeCheck() && filmGateIndex < k_ApertureFormatValues.Length)
-                    cam.sensorSize.vector2Value = k_ApertureFormatValues[filmGateIndex];
+                int oldFilmGateIndex = Array.IndexOf(k_ApertureFormatValues, new Vector2((float)Math.Round(cam.sensorSize.vector2Value.x, 3), (float)Math.Round(cam.sensorSize.vector2Value.y, 3)));
+
+                // If it is not one of the preset sizes, set it to custom
+                oldFilmGateIndex = (oldFilmGateIndex == -1) ? k_CustomPresetIndex: oldFilmGateIndex;
+
+                // Get the new user selection
+                int newFilmGateIndex = EditorGUILayout.Popup(cameraTypeContent, oldFilmGateIndex, k_ApertureFormatNames);
+
+                if (EditorGUI.EndChangeCheck())
+                {
+                    // When switching from custom to a preset, save the last custom value (to display again, in case the user switches back to custom)
+                    if (oldFilmGateIndex == k_CustomPresetIndex)
+                    {
+                        s_LastCustomSensorSizeValue = cam.sensorSize.vector2Value;
+                    }
+
+                    if (newFilmGateIndex != k_CustomPresetIndex)
```

Comment:
Much cleaner thank you! Though I wonder if we should still check with a < , if we change k_ApertureFormatValues at some point, given that k_CustomPresetIndex is defined on the Names vector and not the value one, we might still risk to enter this side of the branch and have issues on k_ApertureFormatValues[newFilmGateIndex];